From c982c90af9c88a7aa159e036722ef18dcf3149da Mon Sep 17 00:00:00 2001
From: Diana Craciun <diana.craciun@oss.nxp.com>
Date: Fri, 16 Oct 2020 12:32:32 +0300
Subject: [PATCH 15/20] vfio/fsl-mc: fix the return of the uninitialized
 variable ret

The vfio_fsl_mc_reflck_attach function may return, on success path,
an uninitialized variable. Fix the problem by initializing the return
variable to 0.

Fixes: f2ba7e8c947b ("vfio/fsl-mc: Added lock support in preparation for interrupt handling")
Reported-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: Diana Craciun <diana.craciun@oss.nxp.com>
Link: https://lore.kernel.org/r/20201016093232.12774-1-diana.craciun@oss.nxp.com
Addresses-Coverity: ("Uninitialized scalar variable")
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index 80fc7f4ed343..0113a980f974 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -58,7 +58,7 @@ static struct vfio_fsl_mc_reflck *vfio_fsl_mc_reflck_alloc(void)
 
 static int vfio_fsl_mc_reflck_attach(struct vfio_fsl_mc_device *vdev)
 {
-	int ret;
+	int ret = 0;
 
 	mutex_lock(&reflck_lock);
 	if (is_fsl_mc_bus_dprc(vdev->mc_dev)) {
-- 
2.29.0.rc1

